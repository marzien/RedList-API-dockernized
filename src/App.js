import React from "react"
import "./App.css"
import { BrowserRouter as Router, Route, Switch } from "react-router-dom"
import RegionList from "./components/RegionList"
import NavBar from "./components/NavBar"
import Mammal from "./components/Mammal"

function App() {
  return (
    <div className="App">
      <NavBar />
      <Router>
        <Switch>
          <Route exact path="/" component={RegionList} />
          <Route path="/mammal" component={Mammal} />
        </Switch>
      </Router>
    </div>
  )
}

export default App
