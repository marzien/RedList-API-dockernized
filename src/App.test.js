import React from "react"
import { render } from "@testing-library/react"
import App from "./App"

test("renders page header correctly", () => {
  const { getByText } = render(<App />)
  const linkElement = getByText(/IUCD RedList API/i)
  expect(linkElement).toBeInTheDocument()
})

test("App snapshot", () => {
  const { container } = render(<App />)
  expect(container.firstChild).toMatchSnapshot()
})
