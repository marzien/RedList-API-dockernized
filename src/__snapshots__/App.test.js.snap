// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  class="App"
>
  <div
    class="container"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        IUCD RedList API
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="mr-auto navbar-nav"
        >
          <a
            class="nav-link"
            data-rb-event-key="/"
            href="/"
          >
            Home
          </a>
        </div>
      </div>
    </nav>
  </div>
  <div>
    <div>
      <div
        class="spinner-border"
        role="status"
      />
      <div>
        Loading...
      </div>
    </div>
  </div>
</div>
`;
